dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -423,49 +484,89 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
 
     void PushNotificationManager::UnregisterAll()
     {
-        THROW_HR_IF(E_NOTIMPL, !::Microsoft::Windows::PushNotifications::Feature_PushNotifications::IsEnabled());
+        bool unregisterCompleted{ false };
+        {
+            auto lock{ m_lock.lock_exclusive() };
+            unregisterCompleted = (!m_singletonForegroundRegistration || !m_comActivatorRegistration);
+        }
 
-        try
+        if (!unregisterCompleted)
         {
             Unregister();
+        }
 
+        {
             auto lock{ m_lock.lock_exclusive() };
+            THROW_HR_IF_MSG(E_FAIL, registering, "Register or Unregister currently in progress!");
```

Comment:
Consider ERROR_OPERATION_IN_PROGRESS instead (or any error code trying to tell the developer that unregistration is in progress, this is the best I found so far 😊).